shared/speaker_setups.hpp

```diff
@@ -0,0 +1,441 @@
+    std::vector<Speaker>{
+      Speaker{"AC_00010001", "L", "M+030", 30.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_00010002", "R", "M-030", -30.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_00010003", "C", "M+000", 0.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_00010004", "LFE", "LFE1", 45.f, -30.f, Layer::bottom, true, false},
+      Speaker{"AC_00010005", "Ls", "M+110", 110.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_00010006", "Rs", "M-110", -110.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_0001000a", "Lss", "M+090", 90.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_0001000b", "Rss", "M-090", -90.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_0001001a", "BL", "M+135", 135.f, 0.f, Layer::middle, false, false}, // This speaker is marked as diffuse
+      Speaker{"AC_0001001b", "BR", "M-135", -135.f, 0.f, Layer::middle, false, false},  // This speaker is marked as diffuse
+      Speaker{"AC_0001000d", "Ltf", "U+030", 30.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_0001000f", "Rtf", "U-030", -30.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_0001000e", "TpFC", "U+000", 0.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_00010010", "Ltr", "U+110", 110.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_00010012", "Rtr", "U-110", -110.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_00010013", "TpSiL", "U+090", 90.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_00010014", "TpSiR", "U-090", -90.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_0001001e", "Ltb", "U+135", 135.f, 30.f, Layer::upper, false, false},
+      Speaker{"AC_0001001f", "Rtb", "U-135", -135.f, 30.f, Layer::upper, false,false}
+    }),
+
+  /* clang-format on */
+};
+
+inline SpeakerSetup speakerSetupByName(const std::string& name) {
+  auto it = std::find_if(
+      SPEAKER_SETUPS.begin(), SPEAKER_SETUPS.end(),
+      [name](auto speakerSetup) { return speakerSetup.name == name; });
+  if (it != SPEAKER_SETUPS.end()) {
+    return *it;
+  } else {
+    return SpeakerSetup{};
+  }
+}
+
+inline SpeakerSetup speakerSetupByCommonName(const std::string& commonName) {
+  auto it = std::find_if(SPEAKER_SETUPS.begin(), SPEAKER_SETUPS.end(),
+                         [commonName](auto speakerSetup) {
+                           return speakerSetup.commonName == commonName;
+                         });
+  if (it != SPEAKER_SETUPS.end()) {
+    return *it;
+  } else {
+    return SpeakerSetup{};
+  }
+}
+
+inline SpeakerSetup speakerSetupByIndex(int index) {
+  if (0 <= index && index < SPEAKER_SETUPS.size()) {
+    return SPEAKER_SETUPS.at(index);
+  } else {
+    return SpeakerSetup{};
+  }
+}
+
+inline SpeakerSetup speakerSetupByPackFormatId(const std::string& packFormatId) {
+  auto it = std::find_if(
+    SPEAKER_SETUPS.begin(), SPEAKER_SETUPS.end(),
+    [packFormatId](auto speakerSetup) { return speakerSetup.packFormatId == packFormatId; });
```

Comment:
maybe
```c++
 [&packFormatId](auto const& speakerSetup) ...
```
I think the [guideline](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in) is pass anything bigger than two int/float/pointers by const ref, otherwise by value, so generally iteratators, built in types etc get passed by value and everything else is `const&`
This is also [recommended](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f52-prefer-capturing-by-reference-in-lambdas-that-will-be-used-locally-including-passed-to-algorithms) for lambda captures when you're using the function inline, except they're const by default.